### ISSUE TITLE
Error: Image not found

### ISSUE THREAD

**Issue by baconcheese113** [google-github-actions/deploy-cloudrun#465]
### TL;DR

I've been using `google-github-actions/auth` and `google-github-actions/deploy-cloudrun` for a year now without issue, but recent deploys are no longer working and giving the error about the Image not being found, also I see in the gcp console that the latest revisions' image url is missing.

### Expected behavior

I would expect the the images are correctly built from my source code and deployed.

### Observed behavior

```
ERROR: (gcloud.run.deploy) Revision 'handle-it-server-00042-fow' is not ready and cannot serve traffic. Image 'gcr.io/handleit-349522/handle-it-server:latest' not found.
```

### Action YAML

```yaml
# Name of your workflow.
name: main

# Trigger the workflow manually or after push a commit
on: [push, workflow_dispatch]

env:
  NODE_VERSION: 18.12.1

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: ikalnytskyi/action-setup-postgres@v4
        with:
          username: postgres
          password: postgres
          database: test
          port: 5432

      - uses: actions/checkout@v4.0.0

      - name: Setup Nodejs
        uses: actions/setup-node@v3.8.1
        with:
          node-version: ${{ env.NODE_VERSION }}
          cache: "npm"

      - name: Build and install
        run: |
          touch .env.testing
          echo "DATABASE_URL=postgresql://postgres:postgres@localhost:5432/test?schema=public" >> .env.testing
          echo "JWT_SECRET=test_secret" >> .env.testing
          npm install
          npm run build
          npx graphql-codegen
      - name: Push Schema and Seed DB
        run: |
          npm run push-db:test
          npm run seed:test
      - name: Run tests
        run: |
          npm run test
  deploy:
    runs-on: ubuntu-latest
    permissions:
      contents: "read"
      id-token: "write"
    needs: test
    if: github.ref == 'refs/heads/master'
    steps:
      - uses: actions/checkout@v4.0.0

      - name: "Authenticate to Google Cloud"
        uses: "google-github-actions/auth@v1.1.1"
        with:
          workload_identity_provider: "projects/847955048569/locations/global/workloadIdentityPools/my-pool/providers/my-provider"
          service_account: "my-service-account@handleit-349522.iam.gserviceaccount.com"

      - id: "deploy"
        uses: "google-github-actions/deploy-cloudrun@v1.0.2"
        with:
          service: "handle-it-server"
          image: "gcr.io/handleit-349522/handle-it-server:latest"

      - name: "Use output"
        run: 'curl "${{ steps.deploy.outputs.url }}"'
```

### Log output

```text
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/27b70bba-a65e-4aa5-9b41-5fcb12ac355d -f /home/runner/work/_temp/be2a054b-dc74-4a22-ac89-c598bc3b4689
Successfully authenticated
Running: gcloud run deploy handle-it-server --quiet --image gcr.io/handleit-349522/handle-it-server:latest --update-labels managed-by=github-actions,commit-sha=6ca64c923afe5b7970c8d42976ea5ee049c36c3f --platform managed --format json --region us-central1
Error: google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy handle-it-server --quiet --image gcr.io/handleit-349522/handle-it-server:latest --update-labels managed-by=github-actions,commit-sha=6ca64c923afe5b7970c8d42976ea5ee049c36c3f --platform managed --format json --region us-central1`: Deploying container to Cloud Run service [handle-it-server] in project [handleit-349522] region [us-central1]
Deploying...
Creating Revision............................done
Routing traffic....failed
Deployment failed
ERROR: (gcloud.run.deploy) Revision 'handle-it-server-00036-zpd' is not ready and cannot serve traffic. Image 'gcr.io/handleit-349522/handle-it-server:latest' not found.
```

### Additional information

Here's a link to the [job](https://github.com/baconcheese113/handle-it-server/actions/runs/6180806143/job/16777857401)

**Comment by sethvargo** [comment]
Does `gcr.io/handleit-349522/handle-it-server:latest` exist and does the service account have IAM permissions to access it? I don't see any containers: https://console.cloud.google.com/gcr/images/handleit-349522/GLOBAL/handle-it-server?pli=1

**Comment by baconcheese113** [comment]
Thanks for the response, the service account has the correct permissions but the remote image doesn't exist. I was under the assumption that the `deploy-cloudrun` action also built it from my source and published it to the address given with the `latest` tag. Not sure how my workflow was deploying correctly

I changed the action to use the `source` option and added artifact registry permissions and now it's working again like before
```yml
      - name: "Authenticate to Google Cloud"
        uses: "google-github-actions/auth@v1.1.1"
        with:
          workload_identity_provider: "projects/847955048569/locations/global/workloadIdentityPools/my-pool/providers/my-provider"
          service_account: "my-service-account@handleit-349522.iam.gserviceaccount.com"

      - id: "deploy"
        uses: "google-github-actions/deploy-cloudrun@v1.0.2"
        with:
          service: "handle-it-server"
          source: ./
```
I'm a little confused, but I'll close the issue
